sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -396,6 +396,8 @@ case class Or(left: Expression, right: Expression) extends BinaryOperator with P
 
   override def sqlOperator: String = "OR"
 
+  override def nullable: Boolean = left.nullable && right.nullable
```

Comment:
uh... I did not read the original JIRA. 

For foldable expressions, we will eventually evaluate them in the Optimizer. Maybe, here, we do not need to introduce the extra complexity.